pkg/fileutils/fileutils.go

```diff
@@ -1,26 +1,49 @@
 package fileutils
 
 import (
-	log "github.com/Sirupsen/logrus"
 	"path/filepath"
+
+	log "github.com/Sirupsen/logrus"
 )
 
 // Matches returns true if relFilePath matches any of the patterns
+// and isn't excluded by any of the subsequent patterns.
 func Matches(relFilePath string, patterns []string) (bool, error) {
-	for _, exclude := range patterns {
-		matched, err := filepath.Match(exclude, relFilePath)
+
+	var matched = false
+
+	for _, pattern := range patterns {
+
+		var negative bool
+		if pattern != "" && pattern[0] == '!' {
+			negative = true
+			pattern = pattern[1:]
+		}
```

Comment:
Does ! by itself mean anything?  From checking Go's Match() it looks like "" doesn't match anything so I would think "!" shouldn't either. I'm thinking it would be good to add:
```
if pattern == "" {
  continue
}
```
after the above if statement.